tla-typechecker/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -45,7 +45,12 @@ class ToEtcExpr(
         val operType = OperT1(Seq(BoolT1), BoolT1)
         val application = mkUniqApp(Seq(operType), this(d.body))
         // We have to introduce a lambda abstraction, as the type checker is expecting this form.
-        mkLet(BlameRef(d.ID), "__Assume_" + d.ID, mkAbs(ExactRef(d.ID), application), inScopeEx)
+        mkLet(
+            BlameRef(d.ID),
+            "__Assume_" + d.definedName.getOrElse(d.ID.toString),
```

Comment:
Should we try to utilize the defined name here, or leave it as is?

Reply:
Let's use the name attribute!

```suggestion
            d.name,
```